packages/abi-coder/src/FunctionFragment.ts

```diff
@@ -221,4 +221,14 @@ export class FunctionFragment<
 
     return coder.decode(bytes, 0) as [DecodedValue | undefined, number];
   }
+
+  /**
+   * Checks if the function is read-only i.e. it only reads from storage, does not write to it.
+   *
+   * @returns True if the function is read-only, false otherwise.
+   */
+  isReadOnly(): boolean {
+    const storageAttribute = this.attributes.find((attr) => attr.name === 'storage');
+    return !storageAttribute?.arguments.includes('write');
```

Comment:
If `arguments` is _undefined_, this will cause a runtime error, is that an unrealistic edge case?

Reply:
That's correct, but if the `storageAttribute` is `undefined` and there is no storage, then should it return `true` or should it throw an error that there is no `storageAttribute` ?

Reply:
@maschad that's a good point. Maybe in that case I should return `undefined`?

Reply:
It depends, I am not as familiar with Sway yet, but from my understanding in pure and view functions in solidity would be read-only, so they wouldn't have a storage attribute, if the case is different in Sway, then I think throwing an error would actually be more appropriate otherwise you can return `true`

Reply:
@maschad https://github.com/FuelLabs/fuels-ts/pull/2008/commits/ec413f537cc0dad3de2be532d6466f3e25f873df

I've added a testcase for a pure function. `isReadOnly` will return `true` for a function that does not use storage at all, since the user does not need to `call()` that function. `get()` is enough.

I've also updated the docstring.

Reply:
Understood, I suppose a user would only want to know if it's read only or not.